backend/pcloud/pcloud.go

```diff
@@ -446,14 +445,15 @@ type listAllFn func(*api.Item) bool
 // Lists the directory required calling the user function on each item found
 //
 // If the user fn ever returns true then it early exits with found = true
-func (f *Fs) listAll(ctx context.Context, dirID string, directoriesOnly bool, filesOnly bool, fn listAllFn) (found bool, err error) {
+func (f *Fs) listAll(ctx context.Context, dirID string, directoriesOnly bool, filesOnly bool, recursive bool, fn listAllFn) (found bool, err error) {
 	opts := rest.Opts{
-		Method:     "GET",
-		Path:       "/listfolder",
-		Parameters: url.Values{},
+		Method: "GET",
+		Path:   "/listfolder",
+		Parameters: url.Values{
+			"recursive": []string{map[bool]string{true: "1", false: "0"}[recursive]},
```

Comment:
I found this to be the only one-line alternative to a ternary operator, but I should have opted for a more Go native solution after finding Go doesn't do ternary operators. Changed it to a call to `opts.Parameters.Set` (instead of add).